### PR TITLE
feat: Properly support arg `week_start` in `pl_wday_lubridate()`

### PR DESCRIPTION
**Pull request by Yousa-Mirage**
This PR focuses on `TODO` related to `pl_wday_lubridate()`. It enhances the `pl_wday_lubridate()` functionality in the package to support arbitrary `week_start` values (1~7), instead of being limited to Sunday-only week starts. It also adds tests for edge cases and error handling. Comments and vignettes are updated to reflect the new behavior.

Now `pl_wday_lubridate()` should behave the same as `lubridate::wday()`.

**Enhancements to weekday calculation:**

* Updated `pl_wday_lubridate` to support any `week_start` value (1~7), with weekday calculation using the formula `(polars_weekday - week_start + 7) %% 7 + 1`. Previously, only `week_start = 7` (Sunday) was supported.
* Added strict validation for `week_start`, ensuring it is a whole number between 1 and 7, with clear error messages for invalid input types or values.

**Testing and Documentation updates:**

* Expanded and refactored tests for `wday()` to cover all valid `week_start` values, variable inputs, and error conditions. Added snapshot tests for error messages.
* Updated the vignette to remove the restriction that `week_start` must be 7.
* Removed an old, non-functional `wday` implementation and `TODO` comments.

### CODE REVIEW

File: vignettes/r-and-polars-expressions.Rmd [hunk 1]
```diff
@@ -397,7 +397,7 @@ out <- tribble(
       Package == "`lubridate`" & Function %in% c("`rollbackward`", "`rollback`", "`rollforward`") ~
         "While time zone handling should mimick the behaviour of `lubridate` in most cases, it is possible that Polars errors if rolling back/forward leads to am ambiguous datetime. It is also possible to have some differences in hours/minutes/seconds when converting between Polars and R.",
       Package == "`lubridate`" & Function == "`wday`" ~
-        "Requires `week_start == 7`. If `label = TRUE`, it returns a string variable and not a factor as in `lubridate`.",
+        "If `label = TRUE`, it returns a string variable and not a factor as in `lubridate`.",
```
Comment (etiennebacher) [inline vignettes/r-and-polars-expressions.Rmd]:
```suggestion
        "`week_start` must be an integer in `tidypolars`. If `label = TRUE`, it returns a string variable and not a factor as in `lubridate`.",
```

**Comment by Yousa-Mirage** [comment]
Aha, thank you. At least the previous PR basically alleviated the failures of Windows CI, though it's quite slow now.
